src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandHelper.cs

```diff
@@ -35,14 +38,22 @@ public static void PrepareStream(Stream stream)
             stream.Position = 0;
         }
 
-        public static void ValidateStream(Stream stream)
+        public static bool TryValidateStream(Stream stream, RetireAttachmentParameters parameters)
         {
+            if (parameters is { Flags: RetiredAttachmentFlags.Retired })
```

Comment:
What does this mean, and why when we put a retired attachments from the client side we validate (in debug only) that we have no stream?